_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+- `"branches": ["master", "next", "next-major"]` => `master` will accept `fix`; `next` will accept `fix` and `feat` commits; `next-major` will accept any type of commits.
+
+Those rules are meant to prevent inconsistent state in releases in which the same releases line is "forked".
+
+For example with the config `"branches": ["master", "next"]` and:
+- last release from `master` is `1.1.0`
+- last release from `next` is `2.0.0`
+
+Committing a breaking change to `master` would trigger the release of `2.0.0` that would fail as the release already exists.
+
+In addition when the `master` and `next` have the same commit head (after a merging `next` into `master` for example), the first commit done on `next` must be a breaking change.
+Otherwise that would result in a "fork" situation.
+
+For example with the config `"branches": ["master", "next"]` and:
+- last release from `master` is `2.0.0`
+- last release from `next` is `2.0.0`
+
+A `feat` commit on `master` will release version `2.1.0`.
+Committing a `feat` commit on `next` would trigger the release of `2.1.0` that would fail as the release already exists.
+
+With specific channel names:
+
+```json
+{
+  "release": {
+    "branches": [
+      "master",
+      {"branch": "next", "channel": "preview"}
+    ]
+  }
+}
+```
+
+#### LTS branches
+
+Workflow to release legacy versions.
+
+Ideal for modules maintaining legacy versions and doing bug fixes and features only releases.
+
+Branch: `1.x`, `2.x`, `master` (any number of lts branches are supported)
+
+Channel: `1.x`, `2.x`, `latest` if supported by the release target
+
+##### LTS branches workflow
+
+| Action                                          | Result                                                                                                                  |
+|-------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| **push** to `master`                            | Release on default channel (if supported, otherwise no channel)                                                         |
+| **merge** feature branch to `master`            | Release on default channel (if supported, otherwise no channel)                                                         |
+| **push** to `1.x` or `2.x` branches             | Release on `1.x` or `2.x` channel (if supported, otherwise no channel) if in range, report an error and error otherwise |
+| **merge** any branch to `1.x` or `2.x` branches | Release on `1.x` or `2.x` channel (if supported, otherwise no channel) if in range, report an error and error otherwise |
+
+##### LTS branches configuration
+
+```json
+{
+  "release": {
+    "branches": ["1.x", "2.x", "master"]
+  }
+}
+```
+
+With specific channel names:
+
+```json
+{
+  "release": {
+    "branches": [
+      {"branch": "v1", "range": "1.x", "channel": "v1"},
+      {"branch": "v2", "range": "2.x", "channel": "v2"},
```

Comment:
This could be tedious to maintain, and it would be easy to just forget to add new majors to the list.

Could this be automated? Could the user simply provide the branch name pattern, and rely on `semantic-release` to auto-generate a branch following the pattern for every major version release?

Reply:
That's the goal ultimately, but not until GitHub allow to protect branches by pattern.
Otherwise anyone who can create a branch on the repo would be able to trigger a release.

Reply:
> but not until GitHub allow to protect branches by pattern.

Except my impression is that y'all are building to be SCM agnostic. (Though I definitely agree with your concern about `anyone who can create a branch on the repo would be able to trigger a release.`)

I would like to see a solution in place to ensure that doesn't happen, and to do so in a SCM agnostic way.

I just don't see a way forward except to require projects to specify each branch, and expect that they will take the appropriate steps to secure those branches.

Maybe we can just assume that if you're giving someone access, they won't act maliciously?

Also, maybe `semantic-release` can fail if it detects that the branch for a release already exists?

Reply:
GitHub now allows protecting branches by pattern.

But also, couldn't semantic-release setup the branch protection with API calls if it needed & configured too? Anyway, GitHub supports this now.